src/hooks/useResponsiveLayout.ts

```diff
@@ -0,0 +1,24 @@
+import {ParamListBase, RouteProp, useRoute} from '@react-navigation/native';
+import useWindowDimensions from './useWindowDimensions';
+
+type RouteParams = ParamListBase & {
+    params: {isInRHP?: boolean};
+};
+type ResponsiveLayoutResult = {
+    shouldUseNarrowLayout: boolean;
+};
+/**
+ * Hook to determine if we are on mobile devices or in the RHP
+ */
+export default function useResponsiveLayout(): ResponsiveLayoutResult {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    try {
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        const {params} = useRoute<RouteProp<RouteParams, 'params'>>();
+        return {shouldUseNarrowLayout: isSmallScreenWidth || (params?.isInRHP ?? false)};
```

Comment:
This still feels like a bad practice, should we instead pass the `params` as a prop and use it? 
What do you think @roryabraham

Reply:
This looks ok to me. It's a bit weird but better than the alternative of having to use `useRoute` and pass a prop every time `useResponsive` layout needs to be used. Plus if we gave `isInRHP` a default value, then it would be easy to forget to pass it when it might be different than the default.

So I think this implementation looks good